perception/image_projection_based_fusion/src/pointpainting_fusion/node.cpp

```diff
@@ -0,0 +1,256 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "image_projection_based_fusion/pointpainting_fusion/node.hpp"
+
+#include <image_projection_based_fusion/utils/geometry.hpp>
+#include <image_projection_based_fusion/utils/utils.hpp>
+#include <lidar_centerpoint/centerpoint_config.hpp>
+#include <lidar_centerpoint/preprocess/pointcloud_densification.hpp>
+#include <lidar_centerpoint/ros_utils.hpp>
+#include <lidar_centerpoint/utils.hpp>
+#include <tier4_autoware_utils/geometry/geometry.hpp>
+#include <tier4_autoware_utils/math/constants.hpp>
+
+namespace image_projection_based_fusion
+{
+
+PointpaintingFusionNode::PointpaintingFusionNode(const rclcpp::NodeOptions & options)
+: FusionNode<sensor_msgs::msg::PointCloud2, DetectedObjects>("pointpainting_fusion", options)
+{
+  const float score_threshold =
+    static_cast<float>(this->declare_parameter<double>("score_threshold", 0.4));
+  const float circle_nms_dist_threshold =
+    static_cast<float>(this->declare_parameter<double>("circle_nms_dist_threshold", 1.5));
+  // densification param
+  const std::string densification_world_frame_id =
+    this->declare_parameter("densification_world_frame_id", "map");
+  const int densification_num_past_frames =
+    this->declare_parameter("densification_num_past_frames", 1);
```

Comment:
densification aren't used yet. pls change the default parameter and set `0` to `densification_num_past_frames` if the value is more than 0 with a message.